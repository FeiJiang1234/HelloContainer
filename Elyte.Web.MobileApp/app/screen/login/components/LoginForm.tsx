import React from 'react';
import { Formik } from 'formik';
import { ElButton, ElErrorMessage, ElInput, ElLink } from 'el/components';
import routes from 'el/navigation/routes';
import * as Yup from 'yup';
import { validator } from 'el/utils';
import { Text } from 'native-base';

const validatePasswordMessage =
    'Password length betweent 8 and 18, include alphanumeric and start with letter';

const validationSchema = Yup.object().shape({
    account: Yup.string().required('Please enter your account.'),
    password: Yup.string()
        .required(validatePasswordMessage)
        .matches(validator.strongPasswordPartten, validatePasswordMessage),
});

const initValue = {
    account: '',
    password: '',
};

export default function LoginForm({ onSubmit, loginError }) {
    return (
        <Formik
            initialValues={initValue}
            validationSchema={validationSchema}
            onSubmit={values => onSubmit(values)}>
            {({ handleChange, handleSubmit, errors, setFieldTouched, touched,isSubmitting }) => (
                <>
                    <ElInput
                        name="account"
                        placeholder="Email or Phone number"
                        keyboardType="email-address"
                        textContentType="emailAddress"
                        onBlur={() => setFieldTouched('account')}
                        onChangeText={handleChange('account')}
                        maxLength={100}
                    />
                    <ElErrorMessage error={errors['account']} visible={touched['account']} />

                    <ElInput
                        name="password"
                        placeholder="Password"
                        secureTextEntry
                        textContentType="password"
                        onBlur={() => setFieldTouched('password')}
                        onChangeText={handleChange('password')}
                        maxLength={18}
                    />
                    <ElErrorMessage error={errors['password']} visible={touched['password']} />

                    <Text textAlign="center" my={2}>
                        Forgot the password? Request for a{' '}
                        <ElLink to={routes.ResetPassword}>Password Reset</ElLink> or{' '}
                        <ElLink to={routes.ContactUs}>Contact us</ElLink>
                    </Text>

                    <ElErrorMessage error={loginError} visible={loginError} />
                    <ElButton onPress={handleSubmit} style={{ marginBottom: 8 }} disabled={isSubmitting}>
                        Sign in
                    </ElButton>
                </>
            )}
        </Formik>
    );
}
