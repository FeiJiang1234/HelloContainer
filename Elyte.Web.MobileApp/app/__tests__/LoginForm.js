import { fireEvent, waitFor } from '@testing-library/react-native';
import LoginForm from '../screen/login/components/LoginForm';
import { NavigationContainer } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import { NativeBaseProvider } from 'native-base';

const renderRoot = ui => {
    const inset = {
        frame: { x: 0, y: 0, width: 0, height: 0 },
        insets: { top: 0, left: 0, right: 0, bottom: 0 },
    };
    return render(
        <NativeBaseProvider initialWindowMetrics={inset}>
            <NavigationContainer>{ui}</NavigationContainer>
        </NativeBaseProvider>,
    );
};

test('render login form snapshot', () => {
    // Arrange
    const handleSubmit = jest.fn();
    const login = renderRoot(<LoginForm onSubmit={handleSubmit} />);

    // Assert
    expect(login).toMatchSnapshot();
});

test('test login validate email and password required', async () => {
    // Arrange
    const handleSubmit = jest.fn();

    const { getByText } = renderRoot(<LoginForm onSubmit={handleSubmit} />);

    // Act
    fireEvent.press(getByText('Sign in'));

    // Assert
    await waitFor(() => {
        expect(getByText('Please enter your account.'));
        expect(
            getByText(
                'Password length betweent 8 and 18, include alphanumeric and start with letter',
            ),
        );
    });
});

test('test login validate password length', async () => {
    // Arrange
    const handleSubmit = jest.fn();

    const { getByTestId, getByText } = renderRoot(<LoginForm onSubmit={handleSubmit} />);

    // Act
    const password = getByTestId('password');
    fireEvent.changeText(password, '1234');
    fireEvent.press(getByText('Sign in'));

    // Assert
    await waitFor(() => {
        expect(
            getByText(
                'Password length betweent 8 and 18, include alphanumeric and start with letter',
            ),
        );
    });
});

test('test login call submit with user data', async () => {
    // Arrange
    const fakeUser = { account: 'fred@elyte.com', password: 'aa1234567' };
    const handleSubmit = jest.fn();

    const { getByTestId, getByText } = renderRoot(<LoginForm onSubmit={handleSubmit} />);

    // Act
    fireEvent.changeText(getByTestId('account'), fakeUser.account);
    fireEvent.changeText(getByTestId('password'), fakeUser.password);
    const submitBtn = getByText('Sign in');
    fireEvent.press(submitBtn);

    // Assert
    await waitFor(() => {
        expect(handleSubmit).toHaveBeenCalledTimes(1);
        expect(handleSubmit).toHaveBeenCalledWith(fakeUser);
    });
});
