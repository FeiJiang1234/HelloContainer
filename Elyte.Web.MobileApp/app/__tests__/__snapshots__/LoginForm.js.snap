// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render login form snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "backgroundColor": "#F0F2F7",
        "borderRadius": 15,
        "flexDirection": "row",
        "marginVertical": 10,
        "padding": 6,
      }
    }
  >
    <View
      dataSet={{}}
      isFocused={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F0F2F7",
          "borderColor": "#d4d4d4",
          "borderRadius": 4,
          "borderWidth": 0,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TextInput
        accessible={true}
        autoCapitalize="none"
        autoCorrect={false}
        dataSet={{}}
        editable={true}
        inputAccessoryViewID="InputAccessoryDone"
        keyboardType="email-address"
        name="account"
        nativeID="undefined-input"
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder="Email or Phone number"
        placeholderTextColor="#808A9E"
        secureTextEntry={false}
        style={
          [
            {
              "backgroundColor": "transparent",
              "color": "#171717",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "height": "100%",
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
              "width": "100%",
            },
            {
              "color": "#0c0c0c",
              "fontFamily": "Avenir",
              "fontSize": 16,
            },
          ]
        }
        testID="account"
        textContentType="emailAddress"
      />
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#F0F2F7",
        "borderRadius": 15,
        "flexDirection": "row",
        "marginVertical": 10,
        "padding": 6,
      }
    }
  >
    <View
      dataSet={{}}
      isFocused={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F0F2F7",
          "borderColor": "#d4d4d4",
          "borderRadius": 4,
          "borderWidth": 0,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TextInput
        accessible={true}
        autoCapitalize="none"
        autoCorrect={false}
        dataSet={{}}
        editable={true}
        inputAccessoryViewID="InputAccessoryDone"
        name="password"
        nativeID="undefined-input"
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder="Password"
        placeholderTextColor="#808A9E"
        secureTextEntry={true}
        style={
          [
            {
              "backgroundColor": "transparent",
              "color": "#171717",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "height": "100%",
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
              "width": "100%",
            },
            {
              "color": "#0c0c0c",
              "fontFamily": "Avenir",
              "fontSize": 16,
            },
          ]
        }
        testID="password"
        textContentType="password"
      />
    </View>
  </View>
  <Text
    dataSet={{}}
    style={
      {
        "backgroundColor": undefined,
        "color": "#171717",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 21,
        "marginBottom": 8,
        "marginTop": 8,
        "textAlign": "center",
        "textDecorationLine": undefined,
      }
    }
  >
    Forgot the password? Request for a
     
    <Text
      onPress={[Function]}
      style={
        [
          {
            "color": "#17C476",
          },
          undefined,
        ]
      }
    >
      Password Reset
    </Text>
     or
     
    <Text
      onPress={[Function]}
      style={
        [
          {
            "color": "#17C476",
          },
          undefined,
        ]
      }
    >
      Contact us
    </Text>
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4280235101,
            4278717979,
          ],
          "endPoint": undefined,
          "locations": [
            0,
            0.98,
          ],
          "startPoint": undefined,
        }
      }
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 60,
            "justifyContent": "center",
            "width": "100%",
          },
          {},
          {
            "marginBottom": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
              "fontWeight": "bold",
            },
            {
              "fontSize": 18,
            },
          ]
        }
      >
        Sign in
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</RNCSafeAreaProvider>
`;
